packages/k2-alpine/src/components/SegmentedControl/SegmentedControl.tsx

```diff
@@ -137,11 +138,13 @@ export const SegmentedControl = ({
               <Segment
                 sx={{ paddingVertical: type === 'thin' ? 8 : 12 }}
                 key={index}
-                ratio={textRatios[index]}
+                ratio={dynamicItemWidth ? textRatios : 1 / items.length}
                 text={item}
                 index={index}
                 selectedIndex={selectedSegmentIndex}
-                onTextWidthChange={width => handleTextWidthChange(index, width)}
+                onTextWidthChange={
+                  dynamicItemWidth ? handleTextWidthChange : undefined
+                }
```

Comment:
also, I made it stop running onTextWidthChange on every render. that function measures component layout so it's a bit heavy. now, only when we are dealing with dynamicItemWidth then onTextWidthChange will run. similar for ratio, we are pass a constant number for non dynamic case and a shared value for dynamic case.